tuf/testutils/swizzler.go

```diff
@@ -532,3 +537,141 @@ func (m *MetadataSwizzler) UpdateTimestampHash() error {
 	}
 	return m.MetadataCache.SetMeta(data.CanonicalTimestampRole, metaBytes)
 }
+
+// MutateRoot takes a function that mutates the root metadata - once done, it
+// serializes the root again
+func (m *MetadataSwizzler) MutateRoot(mutate func(*data.Root)) error {
+	signedThing, err := signedFromStore(m.MetadataCache, data.CanonicalRootRole)
+	if err != nil {
+		return err
+	}
+
+	var root data.Root
+	if err := json.Unmarshal(signedThing.Signed, &root); err != nil {
+		return err
+	}
+
+	// get the original keys, in case the mutation messes with the signing keys
+	oldPubKeys, err := getPubKeys(m.CryptoService, signedThing, data.CanonicalRootRole)
+	if err != nil {
+		return err
+	}
+
+	mutate(&root)
+
+	rootBytes, err := json.MarshalCanonical(root)
```

Comment:
Could you elaborate?  Wouldn't that produce a Signed object whose `Signed` field was canonically marshalled?